src/Compilers/Core/Portable/Symbols/TypeCompareKind.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Specifies the different kinds of comparison between types.
+    /// </summary>
+    [Flags]
+    internal enum TypeCompareKind
+    {
+        ConsiderEverything = 0,
+        IgnoreCustomModifiersAndArraySizesAndLowerBounds = 1,
+        IgnoreDynamic = 2,
+        IgnoreTupleNames = 4,
+        AllIgnoreOptions = IgnoreCustomModifiersAndArraySizesAndLowerBounds | IgnoreDynamic | IgnoreTupleNames
```

Comment:
There are many instances of `TypeCompareKind.IgnoreDynamic | TypeCompareKind.IgnoreTupleNames`. Consider adding a `IgnoreDynamcAndTupleNames` value.